packages/mongo/collection.js

```diff
@@ -301,6 +298,24 @@ Object.assign(Mongo.Collection.prototype, {
     }
   },
 
+  // The Mongo Node driver is configured to `ignoreUndefined`
+  // fields in documents/selectors. This function can be used to remove
+  // `undefined` fields, to keep things consistent, and help avoid any
+  // unwarranted side effects (such as issue #9619).
+  _removeUndefinedFields(selector) {
```

Comment:
Let's add a `seen = new Set` parameter here, and pass it through to the recursive call below, so that we can keep track of objects we've seen before, to avoid infinite loops if the selector has any cycles.

Reply:
Great idea - will do, thanks!